### PR TITLE
Gradle plugin

### PR DESCRIPTION
**Pull request by rekire**
Here is the first version of the gradle plugin. you need to publish it to your local maven repo in order to test it.

```shell
./gradlew :gP:pTML
```

If it is yelling around that the plugin is unknown comment out the id and the snabble dsl. after the first execution of the command above it should work without out commenting it. This is basically just because it is not yet published.

### CODE REVIEW

**Comment by rekire** [comment]
The huge changelog is caused by the stupid suggestion of AS to rebase the branch
the documentation changes are not yet done

File: core/src/main/java/io/snabble/sdk/SnabbleInitializer.kt [hunk 1]
```diff
@@ -6,13 +6,68 @@ import android.content.pm.PackageManager
...
+import java.util.*
 
 /**
  * Initializer for the snabble SDK using androidx.startup
  */
 class SnabbleInitializer : Initializer<Snabble> {
     override fun create(context: Context): Snabble {
         val app = context.applicationContext as Application
+        val propertiesFiles = mutableMapOf<Environment, String>()
+        fun Environment.capitalizedName() = Character.toUpperCase(name[0]) + name.substring(1).lowercase()
```
Comment (ajungg) [inline core/src/main/java/io/snabble/sdk/SnabbleInitializer.kt:19]: Should be moved to Environment class

File: core/src/main/java/io/snabble/sdk/SnabbleInitializer.kt [hunk 2]
```diff
@@ -6,13 +6,68 @@ import android.content.pm.PackageManager
...
+        context.resources.assets.list("snabble/")?.forEach { path ->
+            Environment.values().forEach { env ->
+                val file = "config${env.capitalizedName()}.properties"
+                if (path == file) propertiesFiles[env] = "snabble/$path"
+            }
+        }
+
+        fun Properties.getBoolean(key: String, default: Boolean) =
```
Comment (ajungg) [inline core/src/main/java/io/snabble/sdk/SnabbleInitializer.kt:27]: PropertiesUtils.kt ?

File: gradle-plugin/src/main/kotlin/io/snabble/setup/BuildEnvironment.kt [hunk 3]
```diff
@@ -0,0 +1,117 @@
+package io.snabble.setup
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * Configuration of the snabble gradle plugin.
+ */
+open class BuildEnvironment(private val environment: Environment) {
```
Comment (ajungg) [inline gradle-plugin/src/main/kotlin/io/snabble/setup/BuildEnvironment.kt:8]: Is there a way to Avoid the duplicatin of the config file?

**Review by ajungg — CHANGES_REQUESTED** [review]
Changes to Config.kt and BuildEnvironment.kt are both ignored and the persisted value is used, even if no value was specified in the applications build.gradle

Also the sample app currently crashes on fresh installs,

**Pull request merged**